### ISSUE TITLE
Required property at root object not generated as required

### ISSUE THREAD

**Issue by bghgary** [CesiumGS/wetzel#25]
See glTF issue: https://github.com/KhronosGroup/glTF/issues/1064

I think this only happens for the root object. The rest of the required properties seem to work properly.

**Comment by scurest** [comment]
This happens because after the top-level `schema` object goes through `normalizeRequired` [here](https://github.com/AnalyticalGraphicsInc/wetzel/blob/668480165326e16d3ced34c7fa649304470853b9/lib/schema4Resolver.js#L49), `schema.properties.asset.required` is set to true, but when it then goes through `normalizeRequired` a second time [here](https://github.com/AnalyticalGraphicsInc/wetzel/blob/668480165326e16d3ced34c7fa649304470853b9/lib/schema4Resolver.js#L60), the `required` field (which is no longer the list of required properties for an `asset` but the flag for whether the `asset` property of `glTF` is required) is erased, so the generator believes it is not required.

Btw, for this case it would be useful to have a debug file for the final output of the resolver too, eg.

```diff
diff --git a/lib/schema4Resolver.js b/lib/schema4Resolver.js
index d43141e..1ef3157 100644
--- a/lib/schema4Resolver.js
+++ b/lib/schema4Resolver.js
@@ -60,6 +60,11 @@ function resolve(schema, fileName, basePath, ignorableTypes, debugOutputPath) {
             normalizeRequired(referencedSchemas[type].schema);
         }
     }
+    if (null !== debugOutputPath) {
+        fs.writeFileSync(debugOutputPath + ".schema4.final.json", JSON.stringify(schemaClone), function (err) {
+            if (err) { console.log(err); }
+        });
+    }
 
     return {
         schema: schemaClone,
```

**Comment by pjcozzi** [comment]
Thanks @bghgary @scurest!  Contributions welcome. 😄

**Comment by emackey** [comment]
Fixed in #38.  Not a glamourous fix but I wanted it to start working ASAP.  Cleanup welcome if you don't like what I did 😄
